pkg/utils/utils.go

```diff
@@ -720,13 +720,42 @@ func generateRandomGUID() net.HardwareAddr {
 
 func GetNicSriovMode(pciAddress string) (string, error) {
 	glog.V(2).Infof("GetNicSriovMode(): device %s", pciAddress)
+
+	supported, err := isDevLinkSupportedBy(pciAddress)
+	if err != nil {
+		return "", err
+	}
+
+	if !supported {
+		return "", nil
+	}
+
 	devLink, err := netlink.DevLinkGetDeviceByName("pci", pciAddress)
```

Comment:
cant we check if device is found or not error ? 

you should get something like:

```
~# devlink dev show pci/0000:04:00.0
kernel answers: No such device
```

Reply:
Considering the actual warning log message:
```
W0703 12:28:47.813124   58755 utils.go:128] DiscoverSriovDevices(): unable to get device mode 0000:3b:00.0 "no such device"
```

If I understood correctly, the check you are suggesting should be:

```
func GetNicSriovMode(pciAddress string) (string, error) {
	glog.V(2).Infof("GetNicSriovMode(): device %s", pciAddress)
	devLink, err := netlink.DevLinkGetDeviceByName("pci", pciAddress)
	if err != nil {
		if err.Error() == "no such device" {
			return "", nil
		}
		return "", err
	}
	return devLink.Attrs.Eswitch.Mode, nil
}
```

(note the N is not capitalized)

I find it a bit weak, but I recognize the value of efficiency. does it looks good to you?

Reply:
the error should be a syscall.Errno value
(netlink read/writes from/to unix socket)

you can check it
```
if err != nil {
  if errors.Is(err, syscall.ENODEV) {
    // do something
  }
  // do something else
}
```